spec/Section 6 -- Execution.md

```diff
@@ -135,110 +152,30 @@ response map.
 
 ExecuteSelectionSet(selectionSet, objectType, objectValue, variableValues):
 
-  * Initialize {visitedFragments} to be the empty set.
   * Let {groupedFieldSet} be the result of
-    {CollectFields(objectType, selectionSet, visitedFragments, variableValues)}.
+    {CollectFields(objectType, selectionSet, variableValues)}.
   * Initialize {resultMap} to an empty ordered map.
-  * For each {groupedFields} in {groupedFieldSet}:
-    * Let {entryTuple} be {GetFieldEntry(objectType, objectValue, groupedFields, variableValues)}.
-    * If {entryTuple} is not {null}:
-      * Let {responseKey} and {responseValue} be the values of {entryTuple}.
-      * Set {responseValue} as the value for {responseKey} in {resultMap}.
+  * For each {groupedFieldSet} as {responseKey} and {fields}:
+    * Let {fieldName} be the name of the first entry in {fields}.
```

Comment:
I like the phrasing you used for a similar case in CollectFields better:

    Let {fieldName} be the name shared by all fields in {fields}.

I think it makes it more intuitive what's going on.

Another option would be to add a note explaining that all {fields} have the same name.